public/js/util/source-map.js

```diff
@@ -1,29 +1,111 @@
 "use strict";
 
 const { networkRequest } = require("./networkRequest");
-const { SourceMapConsumer } = require("source-map");
+const invariant = require("invariant");
+const { SourceMapConsumer, SourceNode } = require("source-map");
+const { Source } = require("../types");
+const { Task } = require("./task");
+const { getSource } = require("../selectors");
 
-let sourceMaps = new Map();
+let sourceMapConsumers = new Map();
+let sourceNodes = new Map();
+
+function hasConsumer(sourceId) {
+  return sourceMapConsumers.has(sourceId);
+}
+
+function getConsumer(sourceId) {
+  return sourceMapConsumers.get(sourceId);
+}
+
+function getOriginalSources(sourceId) {
+  const consumer = getConsumer(sourceId);
+  return consumer.sources;
+}
+
+function isOriginal(originalSource) {
+  return !!getGeneratedSourceId(originalSource);
```

Comment:
I think that's kind of confusing to see stringified objects when debugging...  It would be just as useful to concat the generatedSourceId and originalSourceId fields into something like `id1-id2`. The rest of the stringified object is just clutter.

The id could simply like `generatedID/source-mapped-sourceID`. Another approach would be to add an `isOriginal` (or `isVirtual` or `isFake`) flag right onto the source object.